draft-ietf-tls-esni.md

```diff
@@ -699,24 +699,30 @@ MAY offer to resume sessions established without ECH.
 ## Client-Facing Server
 
 Upon receiving an "encrypted_client_hello" extension, the client-facing server
-MUST check that it is able to negotiate TLS 1.3 or greater. If not, it MUST
-abort the connection with a "handshake_failure" alert.
+first checks that it is able to negotiate TLS 1.3 or higher. If not, then it
+MUST ignore the extension and proceed with the handshake as usual. Next, the
+server checks that the ClientHello contains a "server_name" extension and that
+the indicated name is the public name of at least one of its ECH configurations.
```

Comment:
What's the reason for checking the public name? It seems the server can just ignore it.

Reply:
See top-level comment. I don't think that works in general and we can resolve #295 much more simply. (The intent was already for this not to be an issue. I just did a bad job of describing the client behavior.)